lib/internal/errors.js

```diff
@@ -1208,7 +1208,10 @@ E('ERR_INVALID_ARG_TYPE',
     if (actual == null) {
       msg += `. Received ${actual}`;
     } else if (typeof actual === 'function' && actual.name) {
-      msg += `. Received function ${actual.name}`;
+      const inspected = lazyInternalUtilInspect()
+        .inspect(actual);
+      msg += `. Received ${inspected.startsWith('[class') ?
+        'class' : 'function'} ${actual.name}`;
```

Comment:
Would this be enough maybe?
```suggestion
      msg += `. Received ${inspected}`;
```

Reply:
Then it would end as `[Function: ...]` and `[class: ...]`, which would be unusual and inconsistent with how the other ones end. (e.g. for the string type, it ends with `type of string ('...')`), what do you think?